src/vm/wren_vm.c

```diff
@@ -488,18 +502,25 @@ static ObjFiber* loadModule(WrenVM* vm, Value name, const char* source)
   }
 
   ObjFn* fn = wrenCompile(vm, module, source, false, true);
+  if(vm->isSourceAllocated)
+  {
+     void * check = vm->config.reallocateFn((void *) source, 0);
+     if(check)
+      return NULL;
+    vm->isSourceAllocated = 0;
```

Comment:
If you look at https://github.com/munificent/wren/blob/02bcefcbe44cdb73efeebc9b2a9702c41a9674a9/src/include/wren.h#L50

The comments describe the way to free an object using wren's allocation functions is to pass the allocated data to the reallocateFn function pointer and give it a size 0. In the defaultReallocate function when the size is 0 it is freed. The reason isSourceAlocated is set to zero afterwards is because it is freed